spec/JustNotSorrySpec.test.js

```diff
@@ -71,131 +59,81 @@ describe('JustNotSorry', () => {
     });
   });
 
-  describe('#handleSearch', () => {
-    let handleSearch;
-
-    beforeEach(() => {
-      handleSearch = jest.spyOn(instance, 'handleSearch');
-    });
-
-    const assertHandlerWasCalled = () => {
-      expect(handleSearch).toHaveBeenCalledTimes(1);
-    };
-
-    afterEach(assertHandlerWasCalled);
-
-    describe('on focus', () => {
-      it('checks for warnings', () => {
-        const node = enterText('just not sorry');
-
-        simulateEvent(node, 'focus');
-
-        expect(wrapper.state('warnings').length).toEqual(2);
-      });
-
-      it('does nothing when given an empty string', () => {
-        const node = enterText();
-
-        simulateEvent(node, 'focus');
-
-        expect(wrapper.state('warnings').length).toEqual(0);
-      });
-    });
+  it('does nothing when given an empty string', async () => {
+    const email = emailContaining('');
 
-    describe('on input', () => {
-      it('updates warnings each time input is triggered', () => {
-        const node = enterText('just not sorry');
+    render(<JustNotSorry />);
+    simulateEvent(email, 'focus');
 
-        simulateEvent(node, 'input');
-      });
+    await waitFor(() => {
+      expect(document.body.getElementsByClassName('jns-warning').length).toBe(
+        0
+      );
     });
+  });
 
-    describe('on cut', () => {
-      it('updates warnings each time input is triggered', () => {
-        const node = enterText('just not sorry');
-
-        simulateEvent(node, 'cut');
+  describe('events handled', () => {
+    const events = ['input', 'focus', 'cut'];
+    for (let i = 0; i < events.length; i++) {
+      const event = events[i];
+      if (event === 'cut') {
+        //https://www.w3.org/TR/clipboard-apis/#clipboard-event-cut
+        // A synthetic cut event can be manually constructed and dispatched,
+        // but it will not affect the contents of the document or of the system clipboard.
```

Comment:
Not ideal...

Manually I can test this still works but I believe by this being a synthetic event and not affecting the system clipboard it doesn't trigger our registered event listener to be invoked. I may also be misunderstanding this but have ran out of all ideas. I also tried using userEvent from react testing library but that wasn't working either.

I'm open to any and all suggestions for this.